_plugins/rosindex_generator.rb

```diff
@@ -666,30 +673,38 @@ def generate_sorted_paginated(site, elements_sorted, default_sort_key, n_element
     end
   end
 
-
   def sort_repos(site)
     repos_sorted = {}
     repos_sorted['name']= @repo_names.sort_by { |name, instances| name }
 
-    repos_sorted['time'] = repos_sorted['name'].reverse.sort_by { |name, instances|
-      instances.default.snapshots.reject { |d,s|
-        not $recent_distros.include?(d)
-      }.map { |d,s|
-        s.data['last_commit_time'].to_s
-      }.max
-    }.reverse
+    repos_sorted['time'] = Hash.new
+    $all_distros.each do |distro|
+      repos_sorted['time'][distro] = repos_sorted['name'].reverse.sort_by { |name, instances|
+        instances.default.snapshots.reject { |d,s|
+          s.nil? or not $recent_distros.include?(d) or not distro == d
+        }.map { |d,s|
+          s.data['last_commit_time'].to_s
+        }.max.to_s
+      }.reverse
+    end
 
-    repos_sorted['doc'] = repos_sorted['name'].reverse.sort_by { |name, instances|
-      -(instances.default.snapshots.count { |d,s|
-        $recent_distros.include?(d) and not s.data['readme_rendered'].nil?
-      })
-    }
+    repos_sorted['doc'] = Hash.new
+    $all_distros.each do |distro|
+      repos_sorted['doc'][distro] = repos_sorted['name'].reverse.sort_by { |name, instances|
+        -(instances.default.snapshots.count { |d,s|
+          $recent_distros.include?(d) and not s.data['readmes'].nil? or not d == distro
```

Comment:
@apojomovsky hmm I'm under the impression this condition doesn't have the intended effect. It will count snapshots that do not belong to the distro of interest. Consider `select`ing snaphots instead.